jekyll/_cci2/server-3-install-prerequisites.adoc

```diff
@@ -45,99 +69,322 @@ CircleCI server users: +
+``` 
+. Configure` kubectl` with your your credentials gcloud credentials: 
++
+```sh
+gcloud container clusters get-credentials circleci-server --region europe-west1
+```
+. Verify your cluster: 
++
+```sh
+kubectl cluster-info
+```
+. Create a service account for this cluster: 
++
+```sh
+gcloud iam service-accounts create <YOUR_SERVICE_ACCOUNT_ID> --description="<YOUR_SERVICE_ACCOUNT_DISPLAY_NAME>"  --display-name="<YOUR_SERVICE_ACCOUNT_DISPLAY_NAME>"
+```
+. Get the credentials for the service account: 
++
+```sh
+gcloud iam service-accounts keys create <PATH_TO_STORE_CREDENTIALS> --iam-account <SERVICE_ACCOUNT_ID>@<YOUR_GOOGLE_CLOUD_PROJECT_ID>.iam.gserviceaccount.com
+```
 
-## External Ports
-[.table.table-striped]
-[cols=6*, options="header", stripes=even]
-|===
-| Port number
-| Protocol
-| Direction
-| Source / Destination
-| Use
-| Notes
-
-| 80
-| TCP
-| Inbound
-| End users
-| HTTP web app traffic
-|
-
-| 443
-| TCP
-| Inbound
-| End users
-| HTTP web app traffic
-|
-
-| 8800
-| TCP
-| Inbound
-| Administrators
-| Admin console
-|
-
-| 22
-| TCP
-| Inbound
-| Administrators
-| SSH
-| Only required for the bastion host
-
-| 64535-65535
-| TCP
-| Inbound
-|
-| SSH into builds
-| Only required for the nomad clients.
-|===
+=== Create a new GitHub OAuth app
+Registering and setting up a new GitHub OAuth app for CircleCI server allows for authorization control to your server installation using GitHub OAuth and for updates to GitHub projects/repos using build status information.
+
+. In your browser navigate to **your GitHub instance** > **Settings** > **Developer Settings** > **OAuth Apps** and click the **New OAuth App** button. 
++
+.New GitHub OAuth App
+image::github-oauth-new.png[Screenshot showing setting up a new OAuth app]
+
+. Complete the following fields based on your planned installation: 
+** *Homepage URL*: The URL of your planned CircleCI installation.
+** *Authorization callback URL*: The authorization callback URL will be the URL of your planned CircleCI installation followed by `/auth/github`
+
+. Once completed you will be shown the *Client ID*. Select *Generate a new Client Secret* to generate a Client Secret for your new OAuth App. You will need these values when you configure CircleCI server. 
++
+.Client ID and Secret
+image::github-clientid.png[Screenshot showing GitHub Client ID]
+
+NOTE: If using GitHub Enterprise, you will also need a personal access token and the domain name of your GitHub Enterprise instance.
+
+=== Frontend TLS certificates
+By default, CircleCI server will create self-signed certificates to get you started. In production, you should supply a certificate from a trusted certificate authority. The link:https://letsencrypt.org/[LetsEncrypt] certificate authority, for example, can issue a certificate for free using their link:https://certbot.eff.org/[certbot] tool. In the sections below we cover using Google Cloud DNS and AWS Route53.
```

Comment:
@ryanwohara and @nathanfishcircleci can you check what we say happens by default here to check it's correct - "default" and "recommended" are different if I understand correctly.

Reply:
This is correct :)